pkg/kubectl/apply/primitive_element.go

```diff
@@ -32,3 +34,24 @@ func (e PrimitiveElement) Merge(v Strategy) (Result, error) {
 }
 
 var _ Element = &PrimitiveElement{}
+
+// HasConflict returns ConflictError if primitive element has conflict field.
+// Conflicts happen when either of the following conditions:
+// 1. A field is specified in both recorded and remote values, but does not match.
+// 2. A field is specified in recorded values, but missing in remote values.
+func (e PrimitiveElement) HasConflict() error {
+	if !e.HasRecorded() && !e.HasRemote() {
+		return nil
+	}
```

Comment:
we can take this `if` block out now. It will fall back to the last `return nil` statement and will be consistent with the documentation as well :)

Reply:
Yes, it should be cleaned so that the code logic is aline with documentation.